version: "3"

services:
  db:
    image: postgres:15-alpine
    expose:
      - "5432"
    volumes:
      - ./db/db_data:/var/lib/postgresql/data/
      - ./postgresql/start_sql:/docker-entrypoint-initdb.d/
      - ./postgresql/import_data:/import_data
      - ./postgresql/export_data:/export_data
      - ../datasets:/datasets
    env_file:
      - postgresql/db.env
    networks:
      - backend
    container_name: s21_retail_db
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres -d postgres" ]
      interval: 10s
      timeout: 5s
      retries: 5
    restart: unless-stopped

  nginx:
    image: nginx:latest
    networks:
      - backend
    ports:
      - "1337:80"
    container_name: s21_retail_nginx
    volumes:
      - ./nginx/nginx.conf:/etc/nginx/nginx.conf:ro
    depends_on:
      - application

  application:
    build:
      context: .
      dockerfile: ./Dockerfile
    expose:
      - "8000"
    networks:
      - backend
    container_name: s21_retail_web
    image: s21_web
    volumes:
      - ./logs:/app/logs
      - ./export:/app/export
    depends_on:
      db:
        condition: service_healthy

networks:
  backend:
    driver: bridge
